version: "2"
services:
  qgisserver:
    restart: always
    image: gisquick/qgis-server
    volumes:
      - ./_data/publish:/publish/:ro
    ports:
      - "9000:90"

  django:
    restart: always
    image: gisquick/django:aws
    links:
      - qgisserver
    volumes:
      - ./_data/media:/var/www/gisquick/media/
      - ./_data/data:/var/www/gisquick/data/
      - ./_data/publish:/publish/
    environment:
      - GUNICORN_ERRORLOG=-
      - GUNICORN_ACCESSLOG=-
      - DJANGO_EMAIL_USE_SSL=False
      - DJANGO_EMAIL_USE_TLS=True
      - DJANGO_EMAIL_BACKEND='django.core.mail.backends.smtp.EmailBackend'
      - DJANGO_EMAIL_HOST='smtp.gmail.com'
      - DJANGO_EMAIL_PORT=587
      - DJANGO_EMAIL_HOST_USER='your@mail.com'
      - DJANGO_EMAIL_HOST_PASSWORD='verysecret'

  nginx:
    restart: unless-stopped
#    image: gisquick/nginx
    build: nginx_rain
    links:
      - django
    volumes_from:
      - django:ro
    volumes:
      - /etc/letsencrypt/:/etc/letsencrypt/
      - /opt/subdayprecip-design/openlayers2/:/var/www/webapp/d-rain-point
      - /var/www/html/docs/:/var/www/docs
      - /opt/subdayprecip-design:/opt/subdayprecip-design
      - /opt/grassdata/subdayprecip:/opt/grassdata/subdayprecip
      - /opt/subdayprecip-design/wps/logs:/var/log/pywps
    ports:
      - "443:443"
    environment:
      - NGINX_HOST=rain1.fsv.cvut.cz
